full-stack-practice/client/src/pages/Catalog.tsx

```diff
@@ -0,0 +1,69 @@
+import './Pages.css';
+import { useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+import { toDollars } from '../lib';
+
+export function Catalog() {
+  const [items, setItems] = useState<Product[]>([]);
+  const [error, setError] = useState<unknown>();
+  const [loading, setLoading] = useState(true);
+
+  // use Effect to asynchronously fetch product data from the array of products in data.ts
+  useEffect(() => {
+    async function loadCatalog() {
+      try {
+        const res = await fetch('/api/products');
+        if (!res.ok) throw new Error(`Response error, status ${res.status}`);
+        const result = await res.json();
+        setItems(result);
+      } catch (error) {
+        setError(error);
+      } finally {
+        setLoading(false);
+      }
+    }
+    loadCatalog();
+  }, []);
+
+  // if data is taking a while to load, display a msg to the user
+  if (loading) {
+    return <div>Loading...</div>;
+  }
+
+  // if error occurs while fetching, let user know
+  if (error) {
+    return (
```

Comment:
It would be helpful during debugging to log the error to the console or display the error message.